src/state/_hookLifecycles.ts

```diff
@@ -0,0 +1,19 @@
+import {
+  loadedVaultAccount,
+  loadedVaultDetails,
+  loadedVaultPositions,
+  vaultFormSlippage,
+  vaultFormValidation,
+} from './vaultsLifecycle';
+
+// only hooked selectors which are not subscribed to via other means are required here
+// so anything these depend on will be initialized when these initialize
+const allTopLevelHookedSelectors = [
+  loadedVaultDetails,
+  loadedVaultPositions,
+  loadedVaultAccount,
+  vaultFormSlippage,
+  vaultFormValidation,
+];
```

Comment:
Since we run `initHookLicecycles` in `main.tsx`. All of the async requests for vault data will happen on initialization?

Reply:
Yep! Basically as intended, want to have the vaults lifecycle running in the background so there's no loading when you navigate to it. We could have it load on page load but my intent was to structure this the same way abacus does - always living, stored in the state, etc, as a test run for migrating parts of abacus.